src/Internal/DependencyManagement/ExtendedContainer.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * ExtendedContainer class file.
+ *
+ * @package Automattic\WooCommerce\Internal\DependencyManagement
+ */
+
+namespace Automattic\WooCommerce\Internal\DependencyManagement;
+
+use Automattic\WooCommerce\Container;
+use League\Container\Definition\DefinitionInterface;
+
+/**
+ * This class extends the original League's Container object by adding some functionality
+ * that we need for WooCommerce.
+ */
+class ExtendedContainer extends \League\Container\Container {
+
+	/**
+	 * Whitelist of classes that we can register using the container
+	 * despite not belonging to the WooCommerce root namespace.
+	 *
+	 * @var string[]
+	 */
+	private $registration_whitelist = array(
```

Comment:
I'll add that comment, but that's because we don't want old code registered in the container as that old code is (in principle/in the long run at least) prone to deprecation and we want to encourage using new code only whenever possible.